rs/index_server/src/collection_manager.rs

```diff
@@ -273,4 +296,38 @@ impl CollectionManager {
         let hash = hasher.finish();
         hash as u32 % num_workers
     }
+
+    pub async fn get_consumer(&self, collection_name: &str) -> MutexGuard<'_, LogConsumer> {
```

Comment:
```suggestion
    pub async fn get_log_consumer_for_collection(&self, collection_name: &str) -> MutexGuard<'_, LogConsumer> {
```

Reply:
I think a better way is to just have this function subscribe to the topic instead. That's the only usage of this function.